python/ray/autoscaler/aws/example-cloudformation.yaml

```diff
@@ -0,0 +1,52 @@
+# An unique identifier for the head node and workers of this cluster.
+cluster_name: cloudformation
+
+# The maximum number of workers nodes to launch in addition to the head node.
+max_workers: 2
+
+# Cloud-provider specific configuration.
+provider:
+    type: aws
+    region: us-west-2
+    availability_zone: us-west-2a
+    cloud_formation:
+      # AWS CloudFormation stacks to deploy before applying Ray cluster config. Stacks are deployed in the order given.
+      # CloudFormation stack name
+      - stack_name: RayClusterExample
+        # Path to your local CloudFormation template file
```

Comment:
```suggestion
        # Path to your local CloudFormation template file
        # WARNING: Changes to this file won't be applied automatically unless you run `ray up` or `ray submit` with the `--no-config-cache` flag!
```
Since the user specifies a CloudFormation template file path in their cluster launcher config, but this file isn't included in the config file hash used by caching, we could miss updates made to the CloudFormation template unless they disable the config cache. 

Maybe we could also include any CloudFormation template file contents as part of the config file hash in a follow-up PR?

The other alternative would be to allow the CloudFormation template file to be inlined as part of the cluster launcher config, but I'm not particularly inclined to take this path due to (1) a lot of copy-pasting (and manual errors) from existing CloudFormation templates into cluster launcher configs, (2) a lot of cluster launcher configs becoming mainly comprised of CloudFormation template config (making it harder to distinguish Ray config from CloudFormation config), and (3) encouraging a bad design pattern for similar integrations (e.g. CDK) to just continue trying to flatten large external file dependencies in the Cluster Launcher Config.